R/epichains.R

```diff
@@ -0,0 +1,283 @@
+#' Print an [`epichains`] object
+#'
+#' @param x An [`epichains`] object.
+#' @param ... Other parameters passed to [print()].
+#' @return Invisibly returns an [`epichains`]. Called for side-effects.
+#' @export
+print.epichains <- function(x, ...) {
+  format(x, ...)
+}
+
+#' Format method for epichains class
+#'
+#' @param x epichains object
+#' @param ... further arguments passed to or from other methods
+#' @return Invisibly returns an [`epichains`]. Called for printing side-effects.
+#' @export
+format.epichains <- function(x, ...) {
+  # check that x is an epichains object
+  validate_epichains(x)
+
+  # summarise the information stored in x
+  chain_info <- summary(x)
+
+  if (attributes(x)$chain_type == "chains_tree") {
+    writeLines(
+      c(
+        sprintf("`epichains` object"),
+
+        "< tree head (from first known ancestor) >\n"
+        )
+      )
+
+    # print head of the simulation output
+    print(head(x[!is.na(x$ancestor), ]))
+
+    cat("< tree tail >\n")
+
+    # print tail of object
+    print(tail(as.data.frame(x)))
+
+    # print summary information
+    writeLines(
+      c(
+        sprintf("Chains simulated: %s", chain_info[["chains"]]),
+        sprintf(
+          "Unique number of ancestors: %s",
+          chain_info[["unique_ancestors"]]
+        ),
+        sprintf(
+          "Unique number of generations: %s", chain_info[["unique_generations"]]
+        )
+      )
+    )
+
+    # Offer more information to view the full dataset
+    writeLines(sprintf("Use View(<object_name>) to view the full output."))
```

Comment:
`View` doesn't work well in remote sessions - Perhaps instead have
```suggestion
    writeLines(sprintf("Use as.data.frame(<object_name>) to view the full output."))
```

Reply:
I've made the change here c0c6197fa0e2e60640140c3380483560a0a4069e.